### ISSUE TITLE
`commutative` assumption needs to be renamed

### ISSUE THREAD

**Issue by mcpl-sympy** [sympy/sympy#19992]
`is_commutative` and `Q.commutative` is used to indicate whether the expression is commutative with respect to multiplication.
However, commutativity is property of operator, not expression. [https://en.wikipedia.org/wiki/Commutative_property](https://en.wikipedia.org/wiki/Commutative_property)

What current `commutative` indicates is, in fact, that the expression being scalar defined in vector space (or in algebra). I think `is_commutative` can be unified with existing `is_scalar` property.

**Comment by oscarbenjamin** [comment]
> How do you think?

I'm not sure I understand the suggestion. The `is_commutative` attribute is used in many places throughout the core although I'm not sure how useful it is in practice. The most obvious use for it would be handling matrices in an Add/Mul but they and up having their own special Add/Mul classes anyway.

**Comment by mcpl-sympy** [comment]
@oscarbenjamin 
I agree.  Since `Mul` practically represents only scalar multiplication in current SymPy, perhaps `is_commutative` and `Q.commutative` can be removed...

**Comment by Upabjojr** [comment]
FYI, in `sympy.tensor.tensor` there is a way to [manage more complex commutation properties](https://github.com/sympy/sympy/blob/6e4ee3392497a69b0ea453ca84ea53970f8a338b/sympy/tensor/tensor.py#L745) for multiplied tensors.

This goes beyond simple commutativity vs noncommutativity, as it also manages anticommutativity and commutativity between and within groups.

Maybe we could add support for a new multiplication node, say `MulPerm` taking as parameters the allowed permutations of the multiplication arguments and the multiplication arguments.

For example, `MulPerm(Permutation(2)(0, 1)(2, 3), [x, y, z, t])` means that `x*y*z*t == y*x*z*t == x*y*t*z == y*x*t*z` but it's not equal to `x*t*y*z`, for example.

Take into account that we already have in SymPy an implementation of the Butler-Portugal algorithm for the canonicalization of expressions subject to custom permutations.
